marker_config:
  joint7:  
    marker_ids: [6, 8]
    alt_marker_ids: [-1]
    plane: 'yz'
    parent: 'r_forearm'
    child: 'r_palm_axis'
    fixed_end: null
    actuator: 'joint7'
    joint_children: ['joint8']
    name: 'wrist abduction'
    lim_low: -0.7854
    lim_high: 0.7854
  joint8:
    marker_ids: [6, 8]
    alt_marker_ids: [-1]
    plane: 'xy'
    parent: 'r_palm_axis'
    child: &palm 'r_palm'
    fixed_end: 'joint_r_laser'
    actuator: 'joint8'
    joint_children: ['jointI1', 'jointM1', 'jointR1', 'jointL1', 'jointT0']
    name: 'wrist flexion'
    lim_low: -0.7854
    lim_high: 0.7854
  jointI1:
    marker_ids:  [0, 2]
    alt_marker_ids: [-1]
    plane: &plane 'xz'
    parent: *palm
    child: 'r_Iproximal'
    fixed_end: null
    actuator: 'jointI1'
    joint_children: ['jointI2']
    name: 'index flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointI2:
    marker_ids:  [2, 3]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Iproximal'
    child: 'r_Imiddle'
    fixed_end: null
    actuator: 'jointI1'
    joint_children: ['jointI3']
    name: 'index flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointI3:
    marker_ids:  [3, 4]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Imiddle'
    child: 'r_Itip'
    fixed_end: 'joint_Ibumper'
    actuator: 'jointI1'
    joint_children: []
    name: 'index flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointM1:
    marker_ids:  [14, 15]
    alt_marker_ids: [-1]
    plane: *plane
    parent: *palm
    child: 'r_Mproximal'
    fixed_end: null
    actuator: 'jointM1'
    joint_children: ['jointM2']
    name: 'middle flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointM2:
    marker_ids:  [15, 16]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Mproximal'
    child: 'r_Mmiddle'
    fixed_end: null
    actuator: 'jointM1'
    joint_children: ['jointM3']
    name: 'middle flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointM3:
    marker_ids:  [16, 17]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Mmiddle'
    child: 'r_Mtip'
    fixed_end: 'joint_Mbumper'
    actuator: 'jointM1'
    joint_children: []
    name: 'middle flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointR1:
    marker_ids: [18, 19]
    alt_marker_ids: [-1]
    plane: *plane
    parent: *palm
    child: 'r_Rproximal'
    fixed_end: null
    actuator: 'jointL1R1'
    joint_children: ['jointR2']
    name: 'ring flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointR2:
    marker_ids: [19, 21]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Rproximal'
    child: 'r_Rmiddle'
    fixed_end: null
    actuator: 'jointL1R1'
    joint_children: ['jointR3']
    name: 'ring flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointR3:
    marker_ids: [21, 23]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Rmiddle'
    child: 'r_Rtip'
    fixed_end: 'joint_Rbumper'
    actuator: 'jointL1R1'
    joint_children: []
    name: 'ring flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointL1:
    marker_ids:  [24, 25]
    alt_marker_ids: [-1]
    plane: *plane
    parent: *palm
    child: 'r_Lproximal'
    fixed_end: null
    actuator: 'jointL1R1'
    joint_children: ['jointL2']
    name: 'little flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointL2:
    marker_ids:  [25, 26]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Lproximal'
    child: 'r_Lmiddle'
    fixed_end: null
    actuator: 'jointL1R1'
    joint_children: ['jointL3']
    name: 'little flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointL3:
    marker_ids:  [26, 27]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Lmiddle'
    child: 'r_Ltip'
    fixed_end: 'joint_Lbumper'
    actuator: 'jointL1R1'
    joint_children: []
    name: 'little flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointT0:
    marker_ids: [8, 10]
    alt_marker_ids: [-1]
    plane: 'xy'
    parent: *palm
    child: 'r_thumb_base'
    fixed_end: null
    actuator: 'jointT0'
    joint_children: ['jointT1']
    name: 'thumb abduction'
    lim_low: -0.7854
    lim_high: 0.7854
    virtual_base_tf :
      trans: [0,0,0]
      rot: [ 0.22909005, -0.31920022, -0.53619658] # [ 0.4136552,  -0.26684877, -0.76265598]
  jointT1:
    marker_ids: [10, 11]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_thumb_base'
    child: 'r_Tproximal'
    fixed_end: null
    actuator: 'jointT1'
    joint_children: ['jointT2']
    name: 'thumb flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointT2:
    marker_ids: [11, 12]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Tproximal'
    child: 'r_Tmiddle'
    fixed_end: null
    actuator: 'jointT1'
    joint_children: ['jointT3']
    name: 'thumb flexion'
    lim_low: 0.0
    lim_high: 1.571
  jointT3:
    marker_ids: [12, 13]
    alt_marker_ids: [-1]
    plane: *plane
    parent: 'r_Tmiddle'
    child: 'r_Ttip'
    fixed_end: 'joint_Tbumper'
    actuator: 'jointT1'
    joint_children: []
    name: 'thumb flexion'
    lim_low: 0.0
    lim_high: 1.571

tf_marker_ids:
  forearm: [5]
  palm: [9]

marker_poses:
  0: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  1: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  2: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  3: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  4: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  5: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  6: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  7: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  8: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  9: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  10: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  11: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  12: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  13: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  14: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  15: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  16: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  17: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  18: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  19: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  20: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  21: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  22: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  23: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  24: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  25: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  26: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  27: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  28: 
    xyz: [0,0,0]
    rpy: [0,0,0]
  29: 
    xyz: [0,0,0]
    rpy: [0,0,0]
